lib/iris/fileformats/netcdf/saver.py

```diff
@@ -709,20 +713,27 @@ def write(
         # aux factory in the cube.
         self._add_aux_factories(cube, cf_var_cube, cube_dimensions)
 
-        # Add data variable-only attribute names to local_keys.
-        if local_keys is None:
-            local_keys = set()
-        else:
-            local_keys = set(local_keys)
-        local_keys.update(_CF_DATA_ATTRS, _UKMO_DATA_ATTRS)
-
-        # Add global attributes taking into account local_keys.
-        global_attributes = {
-            k: v
-            for k, v in cube.attributes.items()
-            if (k not in local_keys and k.lower() != "conventions")
-        }
-        self.update_global_attributes(global_attributes)
+        if not iris.FUTURE.save_split_attrs:
+            # In the "old" way, we update global attributes as we go.
+            # Add data variable-only attribute names to local_keys.
+            if local_keys is None:
+                local_keys = set()
+            else:
+                local_keys = set(local_keys)
+            local_keys.update(_CF_DATA_ATTRS, _UKMO_DATA_ATTRS)
+
+            # Add global attributes taking into account local_keys.
+            cube_attributes = cube.attributes
+            if iris.FUTURE.save_split_attrs:
+                # In this case, do *not* promote any 'local' attributes to global ones,
+                # only "global" cube attrs may be written as global file attributes
+                cube_attributes = cube_attributes.globals
```

Comment:
This block can never be reached

Reply:
:+1:
look like that got left over from my previous mods to this code.